lib/pingurl.js

```diff
@@ -12,51 +12,54 @@ const TIMEOUT_MS = 5000;
 const TIMEOUT_WAIT_MS = 1000;
 const MAX_ATTEMPTS = 3;
 
+const sleep = ms => new Promise(r => setTimeout(r, ms));
+const getAgent = (url, a) => (a ? (url.match(/^https/) ? a.https : a.http) : undefined);
+
 /**
  * Attempt to GET a url, with retries/error handling
  */
-exports.ping = async (pingUrl, inputData, timeout, timeoutWait, agents) => {
-  let parsed = url.parse(pingUrl);
+exports.ping = async (url, inputData, timeout, timeoutWait, agents) => {
+  const opts = {
+    method: 'GET',
+    headers: exports.parseHeaders(inputData),
+    agent: getAgent(url, agents),
+  };
+  return exports.request(url, opts, null, timeout, timeoutWait);
+};
+
+/**
+ * Make a request with timeouts/retries
+ */
+exports.request = async (url, opts, data, timeout, timeoutWait) => {
```

Comment:
So for #90, I'll probably add a `exports.post` function to this file.  Which does the same thing as `exports.ping` but has a `opts.method = 'POST'` and provides a request body `data`.